src/components/application_manager/rpc_plugins/app_service_rpc_plugin/test/app_service_plugin/app_service_app_extension_test.cc

```diff
@@ -0,0 +1,220 @@
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "app_service_rpc_plugin/app_service_app_extension.h"
+#include <memory>
+#include "app_service_rpc_plugin/app_service_rpc_plugin.h"
+#include "application_manager/mock_application.h"
+#include "gtest/gtest.h"
+
+namespace {
+const std::string kAppServiceType1 = "AppServiceType1";
+const std::string kAppServiceType2 = "AppServiceType2";
+const std::string kResumptionDataKey = "kResumptionDataKey";
+const std::string kAppServiceInfoKey = "appService";
+}  // namespace
+
+namespace app_service_plugin_test {
+
+using test::components::application_manager_test::MockApplication;
+using ::testing::_;
+using ::testing::Mock;
+using ::testing::NiceMock;
+using ::testing::Return;
+using ::testing::ReturnNull;
+using ::testing::ReturnRef;
+using ::testing::SaveArg;
+
+using namespace app_service_rpc_plugin;
+namespace strings = application_manager::strings;
+namespace plugins = application_manager::plugin_manager;
+
+class AppServiceAppExtensionTest : public ::testing::Test {
+ public:
+  AppServiceAppExtensionTest()
+      : mock_app_(std::make_shared<NiceMock<MockApplication> >())
+      , app_service_app_extension_(
+            std::make_shared<app_service_rpc_plugin::AppServiceAppExtension>(
+                app_service_plugin_, *mock_app_)) {
+    ON_CALL(*mock_app_, AddExtension(_)).WillByDefault(Return(true));
+    ON_CALL(*mock_app_, QueryInterface(_))
+        .WillByDefault(Return(app_service_app_extension_));
+  }
+
+ protected:
+  app_service_rpc_plugin::AppServiceRpcPlugin app_service_plugin_;
+  std::shared_ptr<MockApplication> mock_app_;
+  std::shared_ptr<AppServiceAppExtension> app_service_app_extension_;
+};
+
+TEST_F(AppServiceAppExtensionTest, SubscribeToAppService_SUCCESS) {
+  auto app_service_app_extension =
+      std::make_shared<AppServiceAppExtension>(app_service_plugin_, *mock_app_);
+  EXPECT_TRUE(
+      app_service_app_extension->SubscribeToAppService(kAppServiceType1));
+  const auto& subs = app_service_app_extension->Subscriptions();
+  EXPECT_EQ(1u, subs.size());
+  EXPECT_TRUE(
+      app_service_app_extension->IsSubscribedToAppService(kAppServiceType1));
+}
+
+TEST_F(AppServiceAppExtensionTest,
+       SubscribeToAppService_SubscribeOneAppServiceType_Twice_FAIL) {
+  auto app_service_app_extension =
+      std::make_shared<AppServiceAppExtension>(app_service_plugin_, *mock_app_);
```

Comment:
@YarikMamykin I think this can be done in SetUp and removed from each test case

Reply:
@AKalinich-Luxoft fixed in d3a6bc63c0ded3551421683575a2de6a5850a091